frame/scheduler/src/lib.rs

```diff
@@ -344,7 +461,13 @@ pub mod pallet {
 								let index = Agenda::<T>::decode_len(until).unwrap_or(0);
 								Lookup::<T>::insert(id, (until, index as u32));
 							}
-							Agenda::<T>::append(until, Some(s));
+
+							if Agenda::<T>::try_append(until, s).is_err() {
+								// There is not enough room in the future block where the call
+								// should be re-scheduled. Nothing that we can do about it.
```

Comment:
Log a warning perhaps?

Reply:
I will maybe emit a `Canceled` event here, last time I tried ti somehow did not work.  
But yea, a log would also be useful.